src/main/java/org/matsim/mosaik2/prepare/GetCountData.java

```diff
@@ -0,0 +1,144 @@
+package org.matsim.mosaik2.prepare;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.log4j.Logger;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class GetCountData {
+
+	private static final Logger logger = Logger.getLogger(CreateCounts.class);
+
+	Map<String, CountingData> countData(String filePath1, String filePath2, Map<String, NodeMatcher.MatchedLinkID> nodeMatcher) throws IOException {
+
+		var result1 = readData(filePath1, nodeMatcher);
+		var result2 = readData(filePath2, nodeMatcher);
+
+		for (var entry : result1.entrySet()) {
+
+			result2.put(entry.getKey(), entry.getValue());
+
+		}
+
+		logger.info("###############################################");
+		logger.info("#\t\t\t\t\t\t\t\t\t\t\t\t#");
+		logger.info("#\t\t\t All Counts were imported! \t\t\t#");
+		logger.info("#\t\t\t  " + result2.keySet().size() + " stations were found!\t\t\t#");
+		logger.info("#\t\t\t\t\t\t\t\t\t\t\t\t#");
+		logger.info("###############################################");
+		return result2;
+
+	}
+
+	private Map<String, CountingData> readData(String filePath, Map<String, NodeMatcher.MatchedLinkID> nodeMatcher) throws IOException {
+
+		Map<String, CountingData> data = new HashMap<>();
+
+		try (var reader = new FileReader(filePath)) {
+			try (var parser = CSVFormat.newFormat(';').withAllowMissingColumnNames().withFirstRecordAsHeader().parse(reader)) {
+
+				for (var record : parser) {
+
+					var idR1 = record.get("Zst") + "_R1";
+					var idR2 = record.get("Zst") + "_R2";
+					if (nodeMatcher.containsKey(idR1) && nodeMatcher.containsKey(idR2) &&
+							(record.get("Wotag").trim().equals("2") || record.get("Wotag").trim().equals("3") || record.get("Wotag").trim().equals("4"))) {
+
+						if (!record.get("PLZ_R1").trim().equals("-1") && !record.get("PLZ_R1").trim().equals("0")) {
```

Comment:
please extract statement into method. If possible one method for both directions like

`if (isValid(record, "PLZ_R1"))`